server/src/main/kotlin/org/javacs/kt/SourcePath.kt

```diff
@@ -125,6 +124,9 @@ class SourcePath(
             if (isTemporary) (all().asSequence() + sequenceOf(parsed!!)).toList()
             else all()
         }
+
+        // Creates a shallow copy
+        fun clone(): SourceFile = SourceFile(uri, content, path, parsed, compiledFile, compiledContext, compiledContainer, language, isTemporary)
```

Comment:
Just curious, could we make `SourceFile` a `data class` to get this implementation for free? Or isn't that possible because it is an `inner class`?